### PR TITLE
Update ruby version to 2.7

### PR DESCRIPTION
**Pull request by Chrizpy**

### CODE REVIEW

File: lib/twingly/http.rb [hunk 1]
```diff
@@ -75,12 +77,12 @@ def initialize_defaults
...
       # rubocop:disable Metrics/MethodLength
-      def http_response_for(method, *args)
+      def http_response_for(method:, **args)
         response = case method
                    when :get
-                     http_get_response(*args)
+                     http_get_response(**args)
                    when :post
-                     http_post_response(*args)
+                     http_post_response(**args)
```
Comment (roback) [inline lib/twingly/http.rb:85]: I only had to switch from `**` to `*` in order to get rid of the warnings, there seems to be no need for changing `method` into a keyword argument (`method:`).

I tested with the following changes, which didn't give me any warnings in 2.7.1:

```diff
diff --git a/lib/twingly/http.rb b/lib/twingly/http.rb
index 9ac7f9c..07b1629 100644
--- a/lib/twingly/http.rb
+++ b/lib/twingly/http.rb
@@ -52,13 +52,11 @@ module Twingly
       end

       def get(url, params: {}, headers: {})
-        http_response_for(method: :get, url: url,
-                          params: params, headers: headers)
+        http_response_for(:get, url: url, params: params, headers: headers)
       end

       def post(url, body:, headers: {})
-        http_response_for(method: :post, url: url,
-                          body: body, headers: headers)
+        http_response_for(:post, url: url, body: body, headers: headers)
       end

       private
@@ -77,7 +75,7 @@ module Twingly
       end

       # rubocop:disable Metrics/MethodLength
-      def http_response_for(method:, **args)
+      def http_response_for(method, **args)
         response = case method
                    when :get
                      http_get_response(**args)
```

Comment (Chrizpy) [reply lib/twingly/http.rb:85]: Yep you're correct, there was no need to change `method`... I don't remember how I reasoned there. 🤔

**Comment by walro** [comment]
Perhaps we should do something about https://github.com/twingly/twingly-http/blob/d5dc1f6482c7825310c0d446ffc3f9ec88927c73/twingly-http.gemspec#L17 while at it. I.e. allowing faraday >= 1.0.

IIRC you must get to >= 1.0 to get rid of all 2.7 warnings.

**Comment by walro** [comment]
Oh, possibly we want to bump `faraday_middleware` too

**Comment by Chrizpy** [comment]
> Perhaps we should do something about
> 
> https://github.com/twingly/twingly-http/blob/d5dc1f6482c7825310c0d446ffc3f9ec88927c73/twingly-http.gemspec#L17
> 
> while at it. I.e. allowing faraday >= 1.0.
> IIRC you must get to >= 1.0 to get rid of all 2.7 warnings.

Code breaks when I bump it up to >= 1.0, gonna investigate 🤔

**Review by roback — APPROVED** [review]
LGTM!

**Comment by roback** [comment]
We should also release a new version of the `twingly-http` gem, as per the instructions at https://github.com/twingly/twingly-http#release-workflow

I think just bumping the patch version is enough for the changes made here.

**Comment by walro** [comment]
I got a mail it's released, but Change log is not updated?

**Comment by Chrizpy** [comment]
See: 49081a730e060884a5b5e6723a24a7834cc71abd :)

**Pull request merged**